packages/tools/replay-tool/src/helpers.ts

```diff
@@ -0,0 +1,267 @@
+        type: "Unknown DDS",
+        snapshotFormatVersion: "1.0",
+        packageVersion: "1.0",
+    };
+
+    async load(
+        runtime: IFluidDataStoreRuntime,
+        id: string,
+        services: IChannelServices,
+        channelAttributes: Readonly<IChannelAttributes>,
+    ): Promise<IChannel> {
+        return new UnknownChannel(id, channelAttributes, services);
+    }
+
+    create(runtime: IFluidDataStoreRuntime, id: string): IChannel {
+        throw new Error("Not implemented");
+    }
+}
+
+class ObjectRegistryWithUnknownChannels implements ISharedObjectRegistry {
+    private static readonly types = new Set<string>();
+
+    constructor(private readonly base: ISharedObjectRegistry) {}
+    public get(name: string): IChannelFactory | undefined {
+        const res = this.base.get(name);
+        if (res) {
+            return res;
+        }
+        if (!ObjectRegistryWithUnknownChannels.types.has(name)) {
+            ObjectRegistryWithUnknownChannels.types.add(name);
+            console.error(`DDS of type ${name} can't be created`);
+        }
+        return new UnknownChannelFactory();
+    }
+}
+
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
+function mixinDataStoreWithAnyChannel(
+    Base: typeof FluidDataStoreRuntime = FluidDataStoreRuntime)
+{
+    return class RuntimeWithRequestHandler extends Base {
+        constructor(
+            dataStoreContext: IFluidDataStoreContext,
+            sharedObjectRegistry: ISharedObjectRegistry,
+        ) {
+            super(dataStoreContext, new ObjectRegistryWithUnknownChannels(sharedObjectRegistry));
+        }
+    } as typeof FluidDataStoreRuntime;
+}
+
+/**
+ * URL Resolver object
+ */
+class ContainerUrlResolver implements IUrlResolver {
+    constructor(private readonly cache?: Map<string, IResolvedUrl>) {
+    }
+
+    public async resolve(request: IRequest): Promise<IResolvedUrl> {
+        if (!this.cache.has(request.url)) {
+            return Promise.reject(new Error(`ContainerUrlResolver can't resolve ${request}`));
+        }
+        return this.cache.get(request.url);
+    }
+
+    public async getAbsoluteUrl(
+        resolvedUrl: IResolvedUrl,
+        relativeUrl: string,
+    ): Promise<string> {
+        throw new Error("Not implemented");
+    }
+}
+
+export async function loadContainer(
+    documentServiceFactory: IDocumentServiceFactory,
+    documentName: string,
+    logger?: TelemetryLogger,
+): Promise<Container> {
+    const resolved: IFluidResolvedUrl = {
+        endpoints: {
+            deltaStorageUrl: "replay.com",
+            ordererUrl: "replay.com",
+            storageUrl: "replay.com",
+        },
+        tokens: {},
+        type: "fluid",
+        url: `fluid-file://localhost:6000/fluid/${documentName}`,
+    };
+
+    const urlResolver = new ContainerUrlResolver(
+        new Map<string, IResolvedUrl>([[resolved.url, resolved]]));
+    const chaincode = new API.Chaincode(
```

Comment:
can we not use client api, should be trivial to make a static code loder that returns a runtimefactory

Reply:
I agree. But since this is replay tool, I don't want to change it too much in this change. This tool needs a lot of other cleanup as well so I would rather do it with that change.